### ISSUE TITLE
unable to create dynamic option 

### ISSUE THREAD

**Issue by jishwanth** [select2/select2#5726]
`
<html>
  <head>

    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/select2/3.5.4/select2.min.css" />

    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
  <div class="container">
  <div class="page-header">
    <h1>select2 with pagination</h1>
    <p>for improved performance with large data sets (20 000 items)</p>
    </div>
        <input id="test" style="width:100%;" placeholder="type a number, scroll for more results" />
    </div>
    
    
    <script type="text/javascript" src='//ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js'></script>
    
    <script src="//cdnjs.cloudflare.com/ajax/libs/lodash.js/4.15.0/lodash.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/select2/3.5.4/select2.min.js"></script>

    <script type="text/javascript">
	
	// Function to shuffle the demo data
	function shuffle(str) {
	  return str
		.split('')
		.sort(function() {
		  return 0.5 - Math.random();
		})
		.join('');
	}

	// For demonstration purposes we first make
	// a huge array of demo data (20 000 items)
	// HEADS UP; for the _.map function i use underscore (actually lo-dash) here
	function mockData() {
	  return _.map(_.range(1, 20000), function(i) {
		return {
		  id: i,
		  text: shuffle('te ststr ing to shuffle') + ' ' + i,
		};
	  });
	}

	(function() {
	  // init select 2
		 $('#test').select2({
			tags:true,
			data: mockData(),
			placeholder: 'search',
			multiple: true,
			// query with pagination
			query: function(q) {
			  var pageSize,
				results,
				that = this;
			  pageSize = 20; // or whatever pagesize
			  results = [];
			  if (q.term && q.term !== '') {
				// HEADS UP; for the _.filter function i use underscore (actually lo-dash) here
				results = _.filter(that.data, function(e) {
				  return e.text.toUpperCase().indexOf(q.term.toUpperCase()) >= 0;
				});
			  } else if (q.term === '') {
				results = that.data;
			  }
			  q.callback({
				results: results.slice((q.page - 1) * pageSize, q.page * pageSize),
				more: results.length >= q.page * pageSize,
			  });
		},
	  });
	})();
		
	
	</script>
  </body>
</html>
`

i have set the parameter tags:true even though iam unable to add dynamic content to select

**Comment by kevin-brown** [comment]
You're on Select2 3.5.4, which was released so long ago that I've forgotten the number of years (3? 4? 5? It's been a while).

Please upgrade to the latest version of Select2.
